src/routers/alpha-router/functions/best-swap-route.ts

```diff
@@ -577,6 +598,52 @@ export async function getBestSwapRouteBy(
             );
         }
       }
+      const v4Routes = bestSwap.filter(
```

Comment:
{v4|v3|v2}routes logic could be simplified with a generic function as all of them seem to do same thing

Reply:
also can comment in non-changed lines, but should we include `&& v3GasModel === undefined` in line `501`?
`if (v2GasModel == undefined && v3GasModel == undefined) {`

Reply:
good catch

Reply:
eventually we should use the gas estimate returned from quoter, and remove all those crazy gas estimate logic inside the best-swap-route function.